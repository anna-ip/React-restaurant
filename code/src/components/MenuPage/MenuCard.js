import React from 'react'
import styles from './MenuCard.module.scss'

const MenuCard = ({ filteredMenu }) => {
  return (
    <>
      {filteredMenu &&
        filteredMenu.map((menu) => (
          <div key={menu.id} className={styles.container}>
            <img
              src={`assets/${menu.img}`}
              alt="dish"
              width="150vw"
              height="150vw"
              className={styles.img}
            />
            <section className={styles.section}>
            <div className={styles.titleContainer}>
            <p className={styles.title}>{menu.title}</p>
            <p>{menu.price}</p>
            </div>
            <p className={styles.desc}>{menu.desc}</p>
            </section>
          </div>
        ))}
    </>
  )
}

export default MenuCard
