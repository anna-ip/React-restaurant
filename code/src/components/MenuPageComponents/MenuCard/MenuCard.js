import React from "react";
import MenuCardItem from "../MenuCardItem/MenuCardItem";

const MenuCard = ({ filteredMenu }) => {
  return (
    <>
      {filteredMenu &&
        filteredMenu.map((menu) => <MenuCardItem key={menu.id} menu={menu} />)}
    </>
  );
};

export default MenuCard;
