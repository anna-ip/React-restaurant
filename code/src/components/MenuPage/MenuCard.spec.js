import React from 'react'
import renderer from 'react-test-renderer'
import MenuCard from './MenuCard'

// renders null ?
test('<MenuCard/> renders correctly', () => {
  const tree = renderer.create(<MenuCard />).toJSON()
  expect(tree).toMatchSnapshot()
})

//? Test image render

//? Test props
//? Test title price desc.
