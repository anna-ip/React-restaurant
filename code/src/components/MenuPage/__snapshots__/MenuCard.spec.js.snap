// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuCard/> renders correctly 1`] = `null`;
